cray_product_catalog/catalog_delete.py

```diff
@@ -61,19 +61,167 @@
+
+    @product_cm.setter
+    def product_cm(self, __product_cm):
+        self.__product_cm = __product_cm
+
+    @property
+    def cm_namespace(self):
+        return self.__cm_namespace
+
+    @cm_namespace.setter
+    def cm_namespace(self, __cm_namespace):
+        self.__cm_namespace = __cm_namespace
+
+    @property
+    def product_name(self):
+        return self.__product_name
+
+    @product_name.setter
+    def product_name(self, __product_name):
+        self.__product_name = __product_name
+
+    @property
+    def product_version(self):
+        return self.__product_version
+
+    @product_version.setter
+    def product_version(self, __product_version):
+        self.__product_version = __product_version
+
+    @property
+    def max_reties_for_main_cm(self):
+        return self.__max_reties_for_main_cm
+
+    @max_reties_for_main_cm.setter
+    def max_reties_for_main_cm(self, __max_reties_for_main_cm):
+        self.__max_reties_for_main_cm = __max_reties_for_main_cm
+
+    @property
+    def max_reties_for_prod_cm(self):
+        return self.__max_reties_for_prod_cm
+
+    @max_reties_for_prod_cm.setter
+    def max_reties_for_prod_cm(self, __max_reties_for_prod_cm):
+        self.__max_reties_for_prod_cm = __max_reties_for_prod_cm
+
+    @property
+    def key(self):
+        return self.__key
+
+    @key.setter
+    def key(self, __key):
+        self.__key = __key
+
+    @property
+    def main_cm_fields(self):
+        return self.__main_cm_fields
+
+    @main_cm_fields.setter
+    def main_cm_fields(self, __main_cm_fields):
+        self.__main_cm_fields = __main_cm_fields
+
+    @property
+    def product_cm_fields(self):
+        return self.__product_cm_fields
+
+    @product_cm_fields.setter
+    def product_cm_fields(self, __product_cm_fields):
+        self.__product_cm_fields = __product_cm_fields
+
+    # private methods
+    def __key_belongs_to_main_cm_fields(self):
+        return self.__key in self.__main_cm_fields
+
+    def __key_belongs_to_prod_cm_fields(self):
+        return self.__key in self.__product_cm_fields
+
+    def __modify_main_cm(self):
+        LOGGER.info("Removing from config_map=%s in namespace=%s for %s/%s (key=%s)",
+                    self.main_cm, self.cm_namespace, self.product_name, self.product_version, self.key)
+        modify_config_map(self.__main_cm, self.__cm_namespace, self.__product_name, self.__product_version,
+                          self.__key, self.__max_reties_for_main_cm,)
+
+    def __modify_product_cm(self):
+        LOGGER.info("Removing from config_map=%s in namespace=%s for %s/%s (key=%s)",
+                    self.product_cm, self.cm_namespace, self.product_name, self.product_version, self.key)
+        modify_config_map(self.__product_cm, self.__cm_namespace, self.__product_name, self.__product_version,
+                          self.__key, self.__max_reties_for_prod_cm,)
+
+    # public method
+    def modify_config_map(self):
```

Comment:
Shall we change this method name as it conflicts with the method name at L225?
Something like `update_config_map`

Reply:
Changing to just modify, as the class name suggest we are modifying the configmap.
`modify_config_map_util.modify()`